rfd/0147-repo-fork-syncing.md

```diff
@@ -0,0 +1,60 @@
+* `sync/upstream-[UPSTREAM_BRANCH_NAME]` - This branch will be an exact copy of the upstream branch.  If there is a need to create a PR in this repo the history should be based off this rather than `main`
+* `sync/rebase` - This branch is where our custom changes will be committed.  It will be rebased on to the upstream changes before being committed into `main`
+ 
+#### Branch Protections
+ 
+The only branch that will be protected is the `sync/rebase` branch.  Other branches are expected to allow force pushes so they can be forcefully synced based off the upstream state.
+ 
+### High Level Implementation
+ 
+At a high level there are two primary components to this implementation:
+1. A GitHub action which will be scheduled as well as triggered on changes to the `sync/rebase` branch.  This action will perform the actual rebase action and commit the results to the respective branches above.
+2. A script that is invoked when the rebase can not be applied cleanly.  This can allow some custom rebase logic as a means to reduce how often the `sync/rebase` branch needs to be manually updated to avoid rebase conflicts.  A simple example is that workflow automations not desired in the repo copy may be removed in the `sync/rebase` branch.  That means that any changes to those committed workflows would result in a conflict.  However it can be safely resolved that any modified files removed in the `sync/rebase` branch can be resolved by simply removing.
+ 
+At a high level the action will do the following steps:
+1. Setup auth and any necessary repo specific configuration
+2. Checkout the latest upstream contents
+3. Fetch our custom changes
+4. Push the exact copy branches `sync/upstream-[UPSTREAM_BRANCH_NAME]` to our fork
+5. Rebase our `sync/rebase` branch on to the latest upstream changes, using the script for resolutions if necessary and possible
+6. Force push the rebased changes to the `main` branch on our repo copy
+ 
+### Teleport Security Scanning
+ 
+This tooling will be leveraged to address the gaps in our release branch scanning by doing the following:
+1. Create 3 repos (1 for each supported version): `teleport-sec_scan-1`, `teleport-sec_scan-2`, `teleport-sec_scan-3`.  `1` will reference our latest release branch, with each subsequent version inspecting a consecutively older version.
+2. Setup the sync automation as documented above
+3. Adjust Dependabot configuration to only notify for security updates on these branches
+4. Adjust CodeQL Configuration to scan on these branches rather than trying to capture all branch coverage within just `teleport`
+ 
+Step 4 is an optional addition, but since Dependabot results will be located here, it seems to co-locate other security reporting to these branches too.  This will also reduce the "Fixed" and "Reopened" flapping that can occur when changes are seen inconsistently between `master` and the release branches.
```

Comment:
Similar to my question above: When dependabot or CodeQL finds issues, who will get alerted and how?  How can we avoid @jentfoo being a single point of failure?

Reply:
We are already monitoring these alerts in our security reports, all new repos are included by default.  I simply planned to not exclude these repos to make sure they are included in the reports.

Let me know if you have any thoughts on how we can improve this further.